library/java/io/envoyproxy/envoymobile/engine/JniLibrary.java

```diff
@@ -97,6 +100,16 @@ public static void load() {
    */
   protected static native int sendData(long stream, ByteBuffer data, boolean endStream);
 
+  /**
+   * Read data from the response stream. Returns immediately.
```

Comment:
nit: about the comment this function does not read data right? it merely informs the envoy mobile client that the platform client is ready for more callbacks on the response path.

As a larger question that just occurred to me, should the function be called something else? readyForData?

Reply:
I think the eventual function will actually take a buffer (and it did in an earlier version of this PR). Also of note, for stream libraries in general, "read" is not unprecedented for an API that simply takes byte count.

Reply:
I'm open to reconsidering/discussing, but for what it's worth, I did consider the naming for a while, and felt that sticking to a stream-focused interface seemed the best model moving forward.